src/main/java/com/google/firebase/auth/AbstractFirebaseAuth.java

```diff
@@ -1186,6 +1186,86 @@ protected Void execute() throws FirebaseAuthException {
     };
   }
 
+  /**
+   * Creates a new provider SAML Auth config with the attributes contained in the specified {@link
+   * SamlProviderConfig.CreateRequest}.
+   *
+   * @param request A non-null {@link SamlProviderConfig.CreateRequest} instance.
+   * @return An {@link SamlProviderConfig} instance corresponding to the newly created provider
+   *     config.
+   * @throws NullPointerException if the provided request is null.
+   * @throws FirebaseAuthException if an error occurs while creating the provider config.
+   */
+  public SamlProviderConfig createSamlProviderConfig(
+      @NonNull SamlProviderConfig.CreateRequest request) throws FirebaseAuthException {
+    return createSamlProviderConfigOp(request).call();
+  }
+
+  /**
+   * Similar to {@link #createSamlProviderConfig} but performs the operation asynchronously.
+   *
+   * @param request A non-null {@link SamlProviderConfig.CreateRequest} instance.
+   * @return An {@code ApiFuture} which will complete successfully with a {@link SamlProviderConfig}
+   *     instance corresponding to the newly created provider config. If an error occurs while
+   *     creating the provider config, the future throws a {@link FirebaseAuthException}.
+   * @throws NullPointerException if the provided request is null.
+   */
+  public ApiFuture<SamlProviderConfig> createSamlProviderConfigAsync(
+      @NonNull SamlProviderConfig.CreateRequest request) {
+    return createSamlProviderConfigOp(request).callAsync(firebaseApp);
+  }
+
+  private CallableOperation<SamlProviderConfig, FirebaseAuthException>
+      createSamlProviderConfigOp(final SamlProviderConfig.CreateRequest request) {
+    checkNotDestroyed();
+    checkNotNull(request, "Create request must not be null.");
+    final FirebaseUserManager userManager = getUserManager();
```

Comment:
Ideally all argument validation should happen outside the `CallableOperation`. This looks like it would validate the `request` parameters within the `CallableOperation`.

Reply:
Okay, I've moved it for both `createSamlProviderConfigOp` and `createOidcProviderConfigOp`. I've added TODOs for a couple other operations that were starting to get out of scope for this PR.